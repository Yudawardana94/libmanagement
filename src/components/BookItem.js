import {
  Dimensions,
  StyleSheet,
  Text,
  View,
  Image,
  Pressable,
} from 'react-native';
import React from 'react';

import AuthorRender from '../components/AuthorRender';

const {width} = Dimensions.get('screen');

const BookItem = ({book, onHandleNavigation}) => {
  return (
    <View style={styles.bookContainer} key={book.cover_id}>
      <Image
        style={styles.bookImage}
        source={{
          uri: `https://covers.openlibrary.org/b/ID/${book.cover_id}-M.jpg`,
        }}
      />
      <View style={styles.descriptionWrapper}>
        <View>
          <View style={styles.fdRow}>
            <Text style={styles.titleBook}>
              {book?.title?.slice(0, 50)}
              {book?.title?.length > 45 && '...'}
            </Text>
          </View>
          <Text style={styles.fw500}>{book.cover_edition_key}</Text>
          <AuthorRender authorData={book.authors} />
        </View>
        <Pressable
          style={styles.borrowButton}
          onPress={() => onHandleNavigation(book)}>
          <Text style={styles.textBorrowButton}>Borrow</Text>
        </Pressable>
      </View>
    </View>
  );
};

export default BookItem;

const styles = StyleSheet.create({
  fdRow: {flexDirection: 'row'},
  fw500: {fontWeight: '500'},
  bookContainer: {
    backgroundColor: 'aliceblue',
    margin: 6,
    padding: 8,
    flexDirection: 'row',
    borderWidth: 0.5,
    borderColor: '#23232366',
    borderRadius: 6,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  bookImage: {
    width: 90,
    height: 135,
  },
  descriptionWrapper: {
    marginLeft: 6,
    width: width - 125,
    justifyContent: 'space-between',
  },
  titleBook: {
    flex: 1,
    flexWrap: 'wrap',
    fontSize: 18,
    fontWeight: '700',
  },
  borrowButton: {
    marginTop: 8,
    marginRight: 16,
    backgroundColor: 'steelblue',
    paddingVertical: 4,
    borderRadius: 2,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textBorrowButton: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 14,
  },
});
