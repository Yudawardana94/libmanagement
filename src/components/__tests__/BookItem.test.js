import React from 'react';
import {create, act} from 'react-test-renderer';
import BookItem from '../BookItem';

const bookItemMocakData = {
  title: 'el.title',
  cover_id: 'el.cover_id',
  authors: [
    {
      url: 'el.authorsurl',
      name: 'el.authors.name',
    },
  ],
  first_publish_year: 0,
  cover_edition_key: 'el.cover_edition_key',
};
const onHandleNavigation = () => {
  return true;
};
const tree = create(
  <BookItem book={bookItemMocakData} onHandleNavigation={onHandleNavigation} />,
);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});
