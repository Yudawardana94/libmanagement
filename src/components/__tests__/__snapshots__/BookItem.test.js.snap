// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "aliceblue",
      "borderColor": "#23232366",
      "borderRadius": 6,
      "borderWidth": 0.5,
      "elevation": 5,
      "flexDirection": "row",
      "margin": 6,
      "padding": 8,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 4,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://covers.openlibrary.org/b/ID/el.cover_id-M.jpg",
      }
    }
    style={
      Object {
        "height": 135,
        "width": 90,
      }
    }
  />
  <View
    style={
      Object {
        "justifyContent": "space-between",
        "marginLeft": 6,
        "width": 625,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 1,
              "flexWrap": "wrap",
              "fontSize": 18,
              "fontWeight": "700",
            }
          }
        >
          el.title
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontWeight": "500",
          }
        }
      >
        el.cover_edition_key
      </Text>
      <View
        style={
          Object {
            "marginTop": 4,
          }
        }
      >
        <Text>
          Author: 
        </Text>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            el.authors.name
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "steelblue",
          "borderRadius": 2,
          "display": "flex",
          "justifyContent": "center",
          "marginRight": 16,
          "marginTop": 8,
          "paddingVertical": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Borrow
      </Text>
    </View>
  </View>
</View>
`;
